ZFSin/zfs/module/zfs/zfs_vnops.c

```diff
@@ -2074,7 +2075,6 @@ zfs_remove(vnode_t *dvp, char *name, cred_t *cr, caller_context_t *ct,
 		(void) sa_lookup(zp->z_sa_hdl, SA_ZPL_XATTR(zfsvfs),
 		    &xattr_obj_unlinked, sizeof (xattr_obj_unlinked));
 		delete_now = may_delete_now && !toobig &&
-		    !vnode_isinuse(vp,0) && !vn_has_cached_data(vp) &&
```

Comment:
This checks the same condition again as the `may_delete_now` flag. Not sure why? 
I didn't see something which may change this?